wandb/sdk/wandb_setup.py

```diff
@@ -100,6 +103,52 @@ def __init__(
         self._check()
         self._setup()
 
+    def add_active_run(self, run: wandb_run.Run) -> None:
+        """Append a run to the active runs list.
+
+        This must be called when a run is initialized.
+
+        Args:
+            run: A newly initialized run.
+        """
+        if run not in self._active_runs:
+            self._active_runs.append(run)
+
+    def remove_active_run(self, run: wandb_run.Run) -> None:
+        """Remove the run from the active runs list.
+
+        This must be called when a run is finished.
+
+        Args:
+            run: A run that is finished or crashed.
+        """
+        try:
+            self._active_runs.remove(run)
+        except ValueError:
+            pass  # Removing a run multiple times is not an error.
+
+    @property
+    def most_recent_active_run(self) -> wandb_run.Run | None:
+        """The most recently initialized run that is not yet finished."""
+        if not self._active_runs:
+            return None
+
+        return self._active_runs[-1]
+
+    def finish_all_active_runs(self) -> None:
```

Comment:
this will also only be correct in the context of a single process right? or theoretically in each process we will do the same, so i guess it is fine

Reply:
Good point, this only finishes runs in the same process. I think even in the world where there is one run at a time, `finish_previous` only finishes runs the process has attached to, so this is still a consistent extension of that behavior.